### ISSUE TITLE
SCIM2 list users not working when a user has an apostrophe in the username

### ISSUE THREAD

**Issue by janakamarasena** [wso2/product-is#6919]
If the userstore has a username with an apostrophe(') (ex: o'brion) when `/scim2/Users` endpoint is called it will return the following response

```
{
    "schemas": [
        "urn:ietf:params:scim:api:messages:2.0:Error"
    ],
    "detail": "Error occurred while retrieving SCIM user information",
    "status": "500"
}
```

Note: To create users with an apostrophe the `UsernameJavaRegEx` for the relevant userstore will have to be set to something like `[a-zA-Z0-9._'\-|//]{3,30}$`

**Comment by dmxunlimit** [comment]
#6914 similar issue

**Comment by GDRDABARERA** [comment]
Fix is merged to kernel-4.5.x branch and 4.6.x branch. Hence closing.
